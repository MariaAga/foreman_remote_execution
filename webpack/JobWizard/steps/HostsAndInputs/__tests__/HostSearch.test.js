import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { mount } from '@theforeman/test';
import { HostSearch } from '../HostSearch';

const mockStore = configureMockStore();
const store = mockStore({});
describe('HostSearch ', () => {
  it('rendring', () => {
    const component = mount(
      <Provider store={store}>
        <HostSearch value="some query" setValue={jest.fn()} />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
