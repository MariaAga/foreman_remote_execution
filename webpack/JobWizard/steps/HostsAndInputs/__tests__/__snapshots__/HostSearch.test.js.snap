// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HostSearch  rendring 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HostSearch
    setValue={
      [MockFunction] {
        "calls": Array [
          Array [
            "",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    value="some query"
  >
    <div
      className="foreman-search-field"
    >
      <SearchBar
        data={
          Object {
            "autocomplete": Object {
              "id": "hostsSearch",
              "url": "/hosts/auto_complete_search",
              "useKeyShortcuts": true,
            },
            "bookmarks": Object {
              "canCreate": true,
              "documentationUrl": "4.1.5Searching",
              "url": "/api/bookmarks",
            },
            "controller": "hosts",
          }
        }
        initialQuery="some query"
        onSearch={[Function]}
      >
        <div>
          search
        </div>
      </SearchBar>
    </div>
  </HostSearch>
</Provider>
`;
